apps/vx-mark/frontend/src/app.test.tsx

```diff
@@ -9,47 +9,87 @@ import {
 
 import fetchMock from 'fetch-mock';
 import * as React from 'react';
+import { MarkAndPrint } from '@votingworks/types';
 import {
   setElectionInStorage,
   setStateInStorage,
 } from '../test/helpers/election';
-import { fakeMachineConfigProvider } from '../test/helpers/fake_machine_config';
 import { fakeTts } from '../test/helpers/fake_tts';
 import { advanceTimersAndPromises } from '../test/helpers/smartcards';
 import { render } from '../test/test_utils';
 import { App } from './app';
-import { MarkAndPrint } from './config/types';
 import { electionSampleDefinition } from './data';
 import { AriaScreenReader } from './utils/ScreenReader';
+import { createApiMock } from '../test/helpers/mock_api_client';
+
+const apiMock = createApiMock();
 
 beforeEach(() => {
   jest.useFakeTimers();
+  apiMock.mockApiClient.reset();
 });
 
-it('uses the card service and machine config service by default', async () => {
-  fetchMock
-    .get('/machine-config', {
-      machineId: '0002',
-      codeVersion: '3.14',
-    })
-    .get('/card/read', {
-      body: {
-        status: 'error',
-      },
-    });
+afterEach(() => {
+  apiMock.mockApiClient.assertComplete();
+});
+
+it('uses the card service by default', async () => {
+  fetchMock.get('/card/read', {
+    body: {
+      status: 'error',
+    },
+  });
   const hardware = MemoryHardware.buildStandard();
+  apiMock.expectGetMachineConfig({ machineId: '0002', codeVersion: '3.14' });
 
-  render(<App hardware={hardware} />);
+  render(<App hardware={hardware} apiClient={apiMock.mockApiClient} />);
 
   await screen.findByText('Card is Backwards');
   expect(fetchMock.done()).toEqual(true);
 });
 
+it('will throw an error when using default api', async () => {
```

Comment:
What is this test for? I don't quite understand it

Reply:
its really just to get code coverage for the case when the appClient param is not defined in App

Reply:
Ah gotcha. Maybe leave a comment explaining that?